### ISSUE TITLE
Warnings when compiling with Fable that don't happen otherwise

### ISSUE THREAD

**Issue by lfr** [fable-compiler/Fable#2341]
### Description

I get this warning:

> _The instantiation of the generic type 'Codec' is missing and can't be inferred from the arguments or return type of this member. Consider providing a type instantiation when accessing this type, e.g. 'Codec<_,_>'._

From the following line of code:
```fsharp
Extra.empty
|> Extra.withCustom Codec.Encode<FreeText> Codec.Decode<FreeText>
```
Compilation outside Fable does not generate this warning as type information _can_ be inferred.

### Repro code

I have created a [REPL example here](https://fable.io/repl/#?code=LAKALgngDgpgBAWQgSQHZhgJwGYEMDGMAPAOS4A0JARgHxwC8cAlulnoXDKgCaiiSxEEACrR4jfABtcAZxmcecAO5MwAC1BwtzVjgLwkaDHsJEA9lQBW5OBcs0+4MXADCZ7jHykKcanSiYTABuuBhwABQAlAxwUrLyXNzKqhogmtoyYKFM+HAAtjB5VFg6kizwAKKo+O7EJGDKaly+DQBcdIa67HUUfnThVJJm+ADWrfXRjFGOASxg2HAARACkAIKLrrX4AHRVNR5ESKKwNEA&html=DwQgJg9gxgLgngBwKYAIAWMC2AbAfAKGAxwKKQEMwCUVgYBLGbJXAMXICNnWAnc2ctmAB6Bkxb4awTEhjl0MGAgC0SAI4BXegDcAvAHIAwhAB2MJGeUAVREn0oop82YPmAHjGHFsAbgdpyHgBnWV0NGAAzZQAOfWpaGTkUE3IZXQAibXokAHcECB4YdIcnCxgMnPowGDRdMCQsqCRlSuq0ABoUehNGekFlIKhBJF0ARgA6AAZOzG76TA1MAaHmMamZ8jd5xeXhtemUDRCeXexOVZMIdPjgQZ56BBgUIJ4oDIwlIIAuYWEAc0YUDAJnGbjgAC9hDwkPlhABZciKZQAUUGaGEWUKGn6kEwwkw5CC5h4wjA9CJGPoWJxEEw4wAVkFilAeBAgkECvQASZcCI7g8YDdsN0ANYoaHYDJEuDMIJoJCyYpoaERd6KBDfX5AkGM+rC7Q8cYmWTCEwIPEcDTYAkAAUm4wAnONRsIoOzhJbreRxrMQW6maQvBQqIQOBAwHAbmTtF0wBl2FwkLx+HJsNcRNHA2GI4HvLggA&css=Q), but it doesn't seem that REPL displays warnings.

Note that the entire solution is available [here](https://github.com/lfr/FSharp.ValidationBlocks/tree/master/src) and that specific line is in the file [/demo/src/Examples.fs](https://github.com/lfr/FSharp.ValidationBlocks/blob/c2818547793834af15f5a3a6d98b542685b20024/src/demo/src/Examples.fs#L14). The definition of Codec itself can be found in the file [/FSharp.ValidationBlocks.Fable/Thoth.fs](https://github.com/lfr/FSharp.ValidationBlocks/blob/c2818547793834af15f5a3a6d98b542685b20024/src/FSharp.ValidationBlocks.Fable/Thoth.fs#L9).

👉 AFAICT only the signatures are relevant to this issue, not the actual code

### Related information

* Fable version: 2.13.0
* Operating system: Win10

**Comment by alfonsogarciacaro** [comment]
Hmm, strange. Do you see the warning in the IDE or the Fable logs? The REPL does show warnings (see image below) but please note the REPL is currently built with Fable 3 and it seems you're using Fable 2. Unfortunately we're not updating Fable 2 anymore, so I'd recommend [upgrading](https://github.com/MangelMaxime/fulma-demo/pull/43) and checking if the problem is solved 🤞 

![image](https://user-images.githubusercontent.com/8275461/104026915-f8f63380-5209-11eb-98d6-cbd04402b189.png)

**Comment by lfr** [comment]
I actually did that, I thought I had the latest version (because that's what npm said) and someone in Slack told me that the latest release is now a dotnet tool. So I installed that which opened a whole new can of worms, but on the upside this particular warning no longer appears so I suppose I'll just close this. Thanks.
